packages/enzyme-test-suite/test/ReactWrapper-spec.jsx

```diff
@@ -647,6 +651,309 @@ describeWithDOM('mount', () => {
 </Foo>`);
     });
   });
+  describeIf(is('>= 16.8'), 'hooks', () => {
+    describe('useState', () => {
+
+      /*
+        Test Component
+      */
+      function FooCounter({ initialCount = 0 }){
+        const [count, setCount] = useState(+initialCount);
+
+        return (
+          <Fragment>
+              <button className="increment" type="button" onClick={() => setCount( count + 1)}>-</button>
+                <span className="counter">
+                  {count}
+                </span>
+              <button className="decrement" type="button" onClick={() => setCount( count - 1)}>+</button>
+          </Fragment>
+        );
+      };
+      
+      // Test data
+      let wrapper;
+      const initialCount = 5;
+
+      before(() => {
+        wrapper = mount(<FooCounter initialCount={initialCount}/>);
+      });
+
+      it('initial render', () => {
+        expect(wrapper.find('.counter').text()).to.equal(initialCount.toString());
+      });
+
+      it('lets increment', () => {
+        // count: 5
+        wrapper.find('.increment').props().onClick();
+        // should be 6
+        expect(wrapper.find('.counter').text()).to.equal((initialCount + 1).toString());
+      });
+
+      it('now decrement', () => {
+        // count: 6
+        wrapper.find('.decrement').props().onClick();
+        // I dont know why this test case is breaking
+        // seems to be working fine in my project repo
+        // TODO: need to figure out why broken test cases here
+        // should be 5
+        expect(wrapper.find('.counter').text()).to.equal((initialCount).toString());
+      });
+    });
+    describe('useEffect with mount effect', () => {
+
+      // Test data
+      let wrapper;
+      const didMountCOunt = 9;
+
+      /*
+        Test Component
+      */
+      function FooCounterWithMountEffect({ initialCount = 0 }){
+        const [count, setCount] = useState(+initialCount);
+
+        useEffect( ()=> {
+          setCount(didMountCOunt);
+        }, []);
+        return (
+          <Fragment>
+              <span className="counter">
+                {count}
+              </span>
+          </Fragment>
+        );
+      };
+      
+
+      before(() => {
```

Comment:
same here; it's preferred to copy-paste this initialization into each `it`.

Reply:
Done